modules/local/rna_downsample.nf

```diff
@@ -0,0 +1,61 @@
+process RNA_DOWNSAMPLE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::samtools=1.16.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.16.1--h6899075_1' :
+        'quay.io/biocontainers/samtools:1.16.1--h6899075_1' }"
+
+    input:
+    tuple val(meta), path(bam)
+    path(downsample_bed)
+
+    output:
+    tuple val(meta), path("*_downsmapled.bam")                                , emit: bam
+    tuple val(meta), path("*_downsmapled.bam.bai")                            , emit: bai
+    tuple val(meta), path("*_downsmapled.bam"), path("*_downsmapled.bam.bai") , emit: bam_bai
+    path  "versions.yml"                                                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def seed_frac       = (args.contains('-s')) ? '' : '-s 0.001'
+    def num_reads       = (args.contains('-num_reads')) ? '' : '80000000'
+
+    """
+    samtools view -@ $task.cpus -b -U non_select.bam -L ${downsample_bed} ${bam} | samtools view $seed_frac -@ $task.cpus -b -o select.bam
+    samtools index non_select.bam
+    fraction=\$(samtools idxstats non_select.bam | cut -f3 | awk -v ct="$num_reads" 'BEGIN {total=0} {total += \$1} END {print ct/total}')
+    percent=\$(samtools idxstats non_select.bam | cut -f3 | awk -v ct="$num_reads"00 'BEGIN {total=0} {total += \$1} END {print ct/total}')
+    percent=\$(echo \$percent | cut -d. -f1)
+
+    if (( \$(echo "\$percent < 100") ))
+    then
```

Comment:
Is percent necessary. Can't you use the fraction like this  `if (( \$echo "\$fraction < 1 | bc") ))`? By using `bc` you should be able to do float comparison